santosfamilyfoundation/RuinaMain
# [RUIN-34] Allow user to select a specific in-progress report

# Description

This change introduces the ability for the user to select a specific in-progress report. This change involved somewhat significant re-structuring of the backgroundSave class and the Welcome, QuickSurvey, and Home components. Specifically, I modified the backgroundSave class to be a regular class rather than a component and be instantiated with a given filename (`filePath`) and a boolean indicating whether this is an "old file" (`oldFilePath`). These variables are used to indicate to the backgroundSave class that the current report is an old report. This modification allowed me to set the `path` variable for the backgroundSave class to `filePath` if `oldFilePath` is true, or set it to the standard naming convention if `oldFilePath` is true (CrashReport + current date + time). This allows the app to continuously save in-progress reports with the same unique name, rather than different names. I modified the Welcome component to render a dialog to allow the user to select a specific in-progress report and send the selected report name to the QuickSurvey component. I modified the QuickSurvey component to load the given report and send the report filename and whether it is an old report to the Home component. The Home component then instantiates a new instance of the backgroundSave class that names files appropriately.

# Testing
1. Open the app and start a new report.
2. Add a bunch of random vehicles, passengers, etc to the report.
3. Close the app.
4. Reopen the app. Select resume old session. 
5. A dialog box should pop up with a list of all the saved unfinished reports. Scroll to the bottom of the list and select the last report (this should be the report you just created).
6. Click continue on the QuickSurvey screen.
7. Your report should reload successfully and be populated as you expect.
8. Add passengers to one of the vehicles.
9. Close the app.
10. Reopen the app, resume old session, and select the same report from the dialog box.
11. Click continue on the QuickSurvey screen and confirm that the new passengers you add are present in the reloaded report.

# Potential Future Changes
* We might still want to explore finding a different naming convention for reports when we auto-save them because the date+time naming convention isn't super understandable.
* We should explore whether the backgroundSave object can be rewritten to be more modular/globally accessible. Right now, we pass essentially the same information to multiple components.
